src/main/java/ita/softserve/course_evaluation/service/impl/CourseServiceImpl.java

```diff
@@ -1,24 +1,63 @@
 package ita.softserve.course_evaluation.service.impl;
 
 
+import ita.softserve.course_evaluation.dto.CourseDto;
+import ita.softserve.course_evaluation.dto.dtoMapper.CourseDtoMapper;
 import ita.softserve.course_evaluation.entity.Course;
+import ita.softserve.course_evaluation.exception.CourseNotFoundException;
 import ita.softserve.course_evaluation.repository.CourseRepository;
 import ita.softserve.course_evaluation.service.CourseService;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
 
 @Service
 public class CourseServiceImpl implements CourseService {
 
-    @Autowired
-    CourseRepository courseRepository;
+    private final CourseRepository courseRepository;
+    private final CourseDtoMapper courseDtoMapper = new CourseDtoMapper();
+
+
+    public CourseServiceImpl(CourseRepository courseRepository) {
+        this.courseRepository = courseRepository;
+    }
+
+    @Override
+    public Course addCourse(CourseDto course) {
+        String name = course.getCourseName();
+        courseRepository.findByCourseName(name).ifPresent(n -> {
+            throw new RuntimeException(name + ": this course already created");
```

Comment:
It's better to have specific exceptions for this case like CourseAlreadyExistsException